test/e2e/providers/exec/docker.go

```diff
@@ -40,30 +41,40 @@ func RunDockerExecCommand(container string, cmd string, workdir string) (
 		args = append(args, strings.Fields(cmd)...)
 	}
 	dockerCmd := exec.Command("docker", args...)
-	stdoutPipe, err := dockerCmd.StdoutPipe()
-	if err != nil {
-		return 0, "", "", fmt.Errorf("error when connecting to stdout: %v", err)
-	}
-	stderrPipe, err := dockerCmd.StderrPipe()
-	if err != nil {
-		return 0, "", "", fmt.Errorf("error when connecting to stderr: %v", err)
-	}
+
+	var stdoutBytes, stderrBytes bytes.Buffer
+	dockerCmd.Stdout, dockerCmd.Stderr = &stdoutBytes, &stderrBytes
+
 	if err := dockerCmd.Start(); err != nil {
 		return 0, "", "", fmt.Errorf("error when starting command: %v", err)
 	}
+	errCh := make(chan error, 1)
+	go func() {
+		errCh <- dockerCmd.Wait()
+	}()
+	select {
+	case err := <-errCh:
+		if err != nil {
+			if e, ok := err.(*exec.ExitError); ok {
+				return e.ExitCode(), stdoutBytes.String(), stderrBytes.String(), nil
+			}
+			return 0, "", "", err
 
-	stdoutBytes, _ := ioutil.ReadAll(stdoutPipe)
-	stderrBytes, _ := ioutil.ReadAll(stderrPipe)
-
-	if err := dockerCmd.Wait(); err != nil {
-		if e, ok := err.(*exec.ExitError); ok {
-			return e.ExitCode(), string(stdoutBytes), string(stderrBytes), nil
 		}
-		return 0, "", "", err
 	}
 
+	//stdoutBytes, _ := ioutil.ReadAll(stdoutPipe)
```

Comment:
Please remember to remove these when code is finished.

Reply:
thanks, this changed is reverted.